sqlmesh/core/engine_adapter/base.py

```diff
@@ -495,6 +495,16 @@ def _create_table_from_columns(
             kwargs: Optional create table properties.
         """
         table = exp.to_table(table_name)
+
+        if not columns_to_types_all_known(columns_to_types):
+            # It is ok if the columns types are not known if the table already exists and IF NOT EXISTS is set
+            if exists and self.table_exists(table_name):
+                return
+            raise SQLMeshError(
+                f"Cannot create a table from a self-referencing query without knowing the column types. "
```

Comment:
i think it'd be helpful for the user to receive a suggestion on a solution, perhaps suggest a user casts all projections

Reply:
i'm a bit confused why this error message explicitly calls out "self-referencing". this method is called create_table_from_columns, is only ever used for self-referencing queries?

Reply:
Ah thank you for catching